### ISSUE TITLE
Clarification for configs folder?

### ISSUE THREAD

**Issue by KushajveerSingh** [facebookresearch/maskrcnn-benchmark#598]
I am new to this library and wanted some clarification on the configs folder. It contains many .yaml files and some even belong to different datasets like pascal, cityscapes. As far as I understand they are separate models trained on the dataset, which their parent folder has the name of.

But in the README, for the demo.py to change the model we have to go to ../configs/caffe2/ and then select the desired model. Now that folder contains the faster R-CNN models with mask and keypoint models also.

So what is the use of the other yaml files, present in other folders? Also, when we go to MODEL_ZOO.md we can download the weights of models as .pth files. So if I download faster r-cnn R-101-FPN model, where do I save it so the config file loads it.

**Comment by LeviViana** [comment]
The different yaml files contain information for the user to train the architectures. For instance, with [this](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/configs/cityscapes/e2e_faster_rcnn_R_50_FPN_1x_cocostyle.yaml) file you will only load the backbone's weights. Thus, the rest of the architecture will be randomly initialized, which is the desired result if you want to train it.

The `caffe2` weights are those from [Detectron](https://github.com/facebookresearch/Detectron), thus all the weights are provided, which is the desired result if you want to do some test/demos/inferences of the pre-trained architecture as-is.  

Some of the other config files can be seen as a receipt to reproduce the results from Detectron. So, if you want to train an architecture from scratch, you'll train faster (and using less memory) using this repo.

You don't actually need to download the weights manually. But, if you do, normally a simple `torch.load` call will do the job. Regarding the Detectron weights, it's necessary to translate the Detectron's weights names to those from this repo, [this module](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/utils/c2_model_loading.py) helps you to do it.

**Comment by KushajveerSingh** [comment]
Thanks for the clarification.
